### ISSUE TITLE
TODO: Porting dbench from autotest to avocado

### ISSUE THREAD

**Issue by harish-24** [avocado-framework/avocado-misc-tests#13]
I am porting 'dbench' testcase from autotest to avocado.
Source: https://github.com/autotest/autotest-client-tests/blob/master/dbench/

I am using the absolute path for self.datadir as a workaround because of #9 issue

I have a doubt regarding while porting.
1. Autotest uses a require_gcc() to check gcc installation before starting the test.
Is there anything similar to this is in avocado or can we have it as a general function in framework for reuse?

**Comment by apahim** [comment]
Avocado does not have such function. As it seems very simple and specific: 
https://github.com/autotest/autotest/blob/master/client/job.py#L469
You can re-write it in your test like this:

```
...
from avocado.utils import process
...
    def test(self):
        try:
            process.run('which gcc')
        except:
            self.error('I need gcc!')
...
```

Or you can use our software manager instead:

```
...
from avocado.utils import process
...
    def test(self):
            software_manager.install_distro_packages({'fedora': ['gcc']})
...
```

 _It requires `sudo` in order to work._

**Comment by harish-24** [comment]
@apahim Thanks. Have sent a pull request #15 with self.error

**Comment by ldoktor** [comment]
Hello @harish-24 I'd recommend the use of `SoftwareManager`. It's a nice feature and it supports easy usage:

```
from avocado.utils.software_manager import SoftwareManager
sm = SoftwareManager()
if not sm.check_installed("gcc"):
    sm.install("gcc")
```

the `sm.check_installed` does not require root, the `sm.install` fails in case sudo is not configured, but in case it is configured it installs gcc and you can proceed.

Although it's not a strong requirement. Important is to fail nicely in case requirements are not met.

**Comment by harish-24** [comment]
@ldoktor Thought of porting from autotest as it is. But this feature can definitely be added. Thanks.

**Comment by apahim** [comment]
PR is merged. Anyway, feel free to improve the code with a new PR at anytime. Thank you.

**Comment by harish-24** [comment]
Have sent a new PR #16 .
